datastore/modeladapter.go

```diff
@@ -79,6 +79,20 @@ func (db *DB) UpdateAllowedModel(model Model, authorization User) (string, error
 	}
 	model.APIKey = apiKey
 
+	// Get the existing model using the ID
+	m, err := db.getModel(model.ID)
+	if err != nil {
```

Comment:
Could it happen other kind of error? Maybe the check should be more explicit here and check that the db returned error is exactly that the model is not found

Reply:
I did consider that, but the error message would be a fairly raw database error. I thought that the sanitized message would be better.

Reply:
At least you should log that raw db error, just in case it happens, for not to drive crazy searching what has happened